src/irep2/irep2_expr.h

```diff
@@ -2401,6 +2403,10 @@ class mul2t : public mul_expr_methods
   mul2t(const type2tc &type, const expr2tc &v1, const expr2tc &v2)
     : mul_expr_methods(type, mul_id, v1, v2)
   {
+#if 0
```

Comment:
We should perform this checking for the entire irep2! :-)

Reply:
I agree, but would suggest to do this in another PR. These assertion still trigger way too often for me to fix here.